backend/model_server/encoders.py

```diff
@@ -25,6 +35,100 @@
 _RERANK_MODELS: Optional[list["CrossEncoder"]] = None
 
 
+class EmbeddingProvider(Enum):
+    OPENAI = "openai"
+    COHERE = "cohere"
+    VOYAGE = "voyage"
+    VERTEX = "vertex"
+
+
+class CloudEmbedding:
+    def __init__(self, api_key: str, provider: str, model: str | None = None):
+        self.api_key = api_key
+        self.model = model
+        try:
+            self.provider = EmbeddingProvider(provider.lower())
+        except ValueError:
+            raise ValueError(f"Unsupported provider: {provider}")
+        self.client = self._initialize_client()
+
+    def _initialize_client(self) -> Any:
+        if self.provider == EmbeddingProvider.OPENAI:
+            return openai.OpenAI(api_key=self.api_key)
+        elif self.provider == EmbeddingProvider.COHERE:
+            return CohereClient(api_key=self.api_key)
+        elif self.provider == EmbeddingProvider.VOYAGE:
+            return voyageai.Client(api_key=self.api_key)
+        elif self.provider == EmbeddingProvider.VERTEX:
+            credentials = service_account.Credentials.from_service_account_info(
+                json.loads(self.api_key)
+            )
+            project_id = json.loads(self.api_key)["project_id"]
+            vertexai.init(project=project_id, credentials=credentials)
+            return TextEmbeddingModel.from_pretrained(
+                self.model or "text-embedding-004"
```

Comment:
the defaults are a bit scattered everywhere, I would prefer having it cleaned up. Either as default args or somewhere else in a centralized module imported here